### PR TITLE
Fix failure in 'pip installation' CI every time there's a new release

### PR DESCRIPTION
**Pull request by crusaderky**
Fix e.g.
https://github.com/coiled/coiled-runtime/actions/runs/4617936173/jobs/8164769127?pr=746
AssertionError: assert <Version('0.5.10')> == <Version('0.5.9')>

### CODE REVIEW

**Comment by crusaderky** [comment]
Test failures are unrelated

**Review by ncclementi — APPROVED** [review]
LGTM, Thanks @crusaderky

File: tests/runtime/test_build.py [hunk 1]
```diff
@@ -103,16 +103,24 @@ def test_install_dist():
...
 def test_latest_coiled():
     # Ensure `coiled-runtime` installs the latest version of `coiled` by default
+    # This is installed from either conda-forge or pip, depending on which github action
+    # is running this test
     v_installed = Version(coiled.__version__)
 
     # Get latest `coiled` release version from conda-forge
     output = subprocess.check_output(
-        shlex.split("conda search --override-channels --json -c conda-forge coiled")
+        shlex.split("conda search --override-channels --json -c conda-forge")
```
Comment (jrbourbeau) [inline tests/runtime/test_build.py:112]:
```suggestion
        shlex.split("conda search --override-channels --json -c conda-forge coiled")
```

[the remark this answers is not in the document]
Comment (crusaderky) [reply tests/runtime/test_build.py]: I suspect a typical user of coiled would not use the coiled channel, just conda-forge.
The coiled-runtime CI doesn't use the coiled channel, either.
This was the reasoning behind removing it.

[the remark this answers is not in the document]
Comment (ncclementi) [reply tests/runtime/test_build.py]: James, you beat me to it, came here to say this, but yes I actually tried to run it without `coiled` at the end and it hangs and takes forever.

[the remark this answers is not in the document]
Comment (crusaderky) [reply tests/runtime/test_build.py]: Whoops.

**Pull request merged**